contracts/bridge/x/v4/RegistrationV4.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {CoreV4} from "./CoreV4.sol";
+
+contract RegistrationV4 {
+    CoreV4 public immutable imx;
+
+    constructor(address payable _imx) {
+        imx = CoreV4(_imx);
+    }
+
+    function getVersion() external view returns (string memory) {
+        return imx.VERSION();
+    }
+
+    function isRegistered(uint256 starkKey) public view returns (bool) {
+        return imx.getEthKey(starkKey) != address(0);
+    }
+
+    function registerAndWithdrawAll(address ethKey, uint256 starkKey, bytes calldata signature, uint256 assetType)
+        external
+    {
+        if (!isRegistered(starkKey)) {
+            imx.registerEthAddress(ethKey, starkKey, signature);
+        }
+        withdrawAll(uint160(ethKey), starkKey, assetType);
+    }
+
+    function withdrawAll(uint256 ethKey, uint256 starkKey, uint256 assetType) public {
+        uint256 ethKeyBalance = imx.getWithdrawalBalance(ethKey, assetType);
+        uint256 starkKeyBalance = imx.getWithdrawalBalance(starkKey, assetType);
+        require(ethKeyBalance != 0 || starkKeyBalance != 0, "No funds to withdraw");
```

Comment:
This should be an error. Errors should be defined at the top of the file, before storage variables, structs and events.
That is: error NoFundsToWithdraw(uint256 ethKey, uint256 starkKey)
and 
if (ethKeyBalance == 0 && starkKeyBalance == 0) {
    revert NoFundsToWithdraw(ethKey, starkKey);
}